src/components/sidebar/LoginBox.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+import { useInput } from '../../utilities/customHooks';
+import { useUser } from '../UserProvider';
+import { useTheme } from '../ThemeProvider';
+
+export default function LoginBox({ handleClose, onLoginToggle = (f) => f }) {
+  const { theme } = useTheme();
+  const { currentUser, login, logout } = useUser();
+  const [emailProp, resetEmail] = useInput('');
+  const [passwordProp, resetPassword] = useInput('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleLoginSubmit = (event) => {
+    event.preventDefault();
+    const { success, error } = login(emailProp.value, passwordProp.value);
+    setErrorMessage(error);
+    if (success) {
+      resetEmail();
+      resetPassword();
+      onLoginToggle(true);
+      handleClose();
+    }
+  };
+
+  const handleLogoutSubmit = (event) => {
+    event.preventDefault();
+
+    logout();
+    onLoginToggle(false);
+  };
+
+  const loginForm = (
+    <form className="login-box" onSubmit={handleLoginSubmit}>
+      <label htmlFor="email">
+        Email:
+        <input
+          type="email"
+          name=""
```

Comment:
Although we're not submitting the form to a server for processing, have a name associated with the input is still good for consistency and accessibility for screen readers to use it to pair control inputs with a name.

Reply:
That's a very good point! I'll change that.